tests/test_backend.py

```diff
@@ -208,534 +200,536 @@ def test_save_dict(self):
 
-        if isinstance(self.cache.client, herd.HerdClient):
-            default_timeout = self.cache.client._backend.default_timeout
+        if isinstance(default_cache.client, herd.HerdClient):
+            default_timeout = default_cache.client._backend.default_timeout
             herd_timeout = (default_timeout + herd.CACHE_HERD_TIMEOUT) * 1000
-            herd_pack_value = self.cache.client._pack(value, default_timeout)
+            herd_pack_value = default_cache.client._pack(value, default_timeout)
             mocked_set.assert_called_once_with(
-                self.cache.client.make_key(key, version=None),
-                self.cache.client.encode(herd_pack_value),
+                default_cache.client.make_key(key, version=None),
+                default_cache.client.encode(herd_pack_value),
                 nx=False,
                 px=herd_timeout,
                 xx=False,
             )
         else:
             mocked_set.assert_called_once_with(
-                self.cache.client.make_key(key, version=None),
-                self.cache.client.encode(value),
+                default_cache.client.make_key(key, version=None),
+                default_cache.client.encode(value),
                 nx=False,
-                px=self.cache.client._backend.default_timeout * 1000,
+                px=default_cache.client._backend.default_timeout * 1000,
                 xx=False,
             )
 
-    def test_delete(self):
-        self.cache.set_many({"a": 1, "b": 2, "c": 3})
-        res = self.cache.delete("a")
-        self.assertTrue(bool(res))
+    def test_delete(self, default_cache):
+        default_cache.set_many({"a": 1, "b": 2, "c": 3})
+        res = default_cache.delete("a")
+        assert bool(res)
 
-        res = self.cache.get_many(["a", "b", "c"])
-        self.assertEqual(res, {"b": 2, "c": 3})
+        res = default_cache.get_many(["a", "b", "c"])
+        assert res == {"b": 2, "c": 3}
 
-        res = self.cache.delete("a")
-        self.assertFalse(bool(res))
+        res = default_cache.delete("a")
+        assert not bool(res)
 
-    @patch("django_redis.cache.DJANGO_VERSION", (3, 1, 0, "final", 0))
-    def test_delete_return_value_type_new31(self):
+    def test_delete_return_value_type_new31(self, default_cache, mocker):
         """delete() returns a boolean instead of int since django version 3.1"""
-        self.cache.set("a", 1)
-        res = self.cache.delete("a")
-        self.assertEqual(type(res), bool)
-        self.assertTrue(res)
-        res = self.cache.delete("b")
-        self.assertEqual(type(res), bool)
-        self.assertFalse(res)
-
-    @patch("django_redis.cache.DJANGO_VERSION", (3, 0, 1, "final", 0))
-    def test_delete_return_value_type_before31(self):
+        mocker.patch("django_redis.cache.DJANGO_VERSION", new=(3, 1, 0, "final", 0))
+        default_cache.set("a", 1)
+        res = default_cache.delete("a")
+        assert type(res) == bool
```

Comment:
let's keep isinstance for simplicity and consistency 
```suggestion
        assert isinstance(res, bool)
```

Reply:
Agree, I've been using https://github.com/pytest-dev/unittest2pytest to convert larger test classes and didn't thoroughly check results.